File: .cpp/Program/Computer.cpp
#pragma once
#include <bits/stdc++.h>
#include "Device.cpp"
#include "CPU.cpp"
#include "RAM.cpp"
#include "Storage.cpp"
#include "HDD.cpp"
#include "SSD.cpp"
using namespace std;

class Computer : public Device {
    private:
        string OS;
        CPU cpu;
        vector<RAM> ram;
        vector<Storage*> storage;

    public:
        Computer () { }

        Computer (string brand, string model, string OS, CPU cpu, vector<RAM> ram, vector<Storage*> storage) : Device (brand, model) {
            this->OS = OS;
            this->cpu = cpu;
            this->ram = ram;
            this->storage = storage;
        }

        void setOS (string OS) { this->OS = OS; }
        void setCPU (CPU cpu) { this->cpu = cpu; }
        void setRAM (vector<RAM> ram) { this->ram = ram; }
        void setStorage (vector<Storage*> storage) { this->storage = storage; }

        string getOS () { return OS; }
        CPU getCPU () { return cpu; }
        vector<RAM> getRAM () { return ram; }
        vector<Storage*> getStorage () { return storage; }

        virtual void info () override {
            cout << "| Brand             : " << getBrand() << endl;
            cout << "| Model             : " << getModel() << endl;
            cout << "| OS                : " << OS << endl;
            cout << "| CPU               : "; cpu.info();
            for (auto &r : ram) {
                if (&r == &ram[0]) cout << "| RAM               : - ";
                else cout << "|                     - ";
                r.info();
            }
            for (auto s : storage) {
                if (s == storage[0]) cout << "| Storage           : - ";
                else cout << "|                     - ";
                s->info();
            }
        }

        virtual ~Computer () { }
};

File: .py/Program/Computer.py
from Device import Device
from CPU import CPU
from RAM import RAM
from Storage import Storage

class Computer(Device):
    __OS = ""
    __CPU = None
    __RAM = []
    __Storage = []

    def __init__(self, brand="", model="", OS="", CPU=None, RAM=None, Storage=None):
        super().__init__(brand, model)
        self.__OS = OS
        self.__CPU = CPU
        self.__RAM = RAM if RAM is not None else []
        self.__Storage = Storage if Storage is not None else []

    def setOS(self, OS):
        self.__OS = OS

    def setCPU(self, CPU):
        self.__CPU = CPU

    def setRAM(self, RAM):
        self.__RAM = RAM

    def setStorage(self, Storage):
        self.__Storage = Storage

    def getOS(self):
        return self.__OS

    def getCPU(self):
        return self.__CPU

    def getRAM(self):
        return self.__RAM

    def getStorage(self):
        return self.__Storage

    def info(self):
        print(f"| Brand             : {self.getBrand()}")
        print(f"| Model             : {self.getModel()}")
        print(f"| OS                : {self.__OS}")
        print(f"| CPU               : ", end="")
        self.__CPU.info()
        for i, ram in enumerate(self.__RAM):
            if i == 0:
                print("| RAM               : - ", end="")
            else:
                print("|                     - ", end="")
            ram.info()
        for i, storage in enumerate(self.__Storage):
            if i == 0:
                print("| Storage           : - ", end="")
            else:
                print("|                     - ", end="")
            storage.info()